src/inmanta/ast/statements/generator.py

```diff
@@ -523,45 +525,24 @@ def pretty_print(self) -> str:
         )
 
     def _normalize_rhs(self, index_attributes: abc.Set[str]) -> None:
+        assert self.type is not None  # Make mypy happy
         for k, v in self.__attributes.items():
             # don't notify the rhs for index attributes because it won't be able to resolve the reference
             # (index attributes need to be resolved before the instance can be constructed)
```

Comment:
Could you move this comment down to right above the `v.normalize` call?

Reply:
done